.circleci/config.yml

```diff
@@ -79,7 +99,17 @@ commands:
               --var AUTH_CLIENT_ID=${<< parameters.auth_client_id >>} \
               --var AUTH_CLIENT_SECRET=${<< parameters.auth_client_secret >>} \
               --var NEW_RELIC_LICENSE_KEY=${<< parameters.new_relic_license >>} \
+              --var POSTGRES_DB=${<< parameters.postgres_db >>} \
+              --var POSTGRES_HOST=${<< parameters.postgres_host >>} \
+              --var POSTGRES_PASSWORD=${<< parameters.postgres_password >>} \
+              --var POSTGRES_USERNAME=${<< parameters.postgres_username >>} \
               --var SESSION_SECRET=${<< parameters.session_secret >>}
+      - run:
+          name: Run database migrations
```

Comment:
Does `cf push` return only after the new application is running? Is there potential for this `cf run-task` command to run against an old version of the application while the `cf push` above is still running?

Reply:
Fortunately, the `Push application with deployment vars` step has to return with a status code 0 before the next step `Run database migrations` can be started. So we'll always be running our migration task with new migrations.

Reply:
:+1: so `cf push` only returns after the new application has been completely deployed?

Reply:
Yes, in effect, circle ci actually waits for the application to start up. It's actually really cool- it even reports out stats from the instance before the step exits. There's a whole stream from the "push application" step you can see in circle ci. https://app.circleci.com/pipelines/github/adhocteam/Head-Start-TTADP/762/workflows/26cb590c-cbc0-4f9e-9c8c-eaec771ae3af/jobs/3508